src/components/SourceTabs.js

```diff
@@ -142,7 +142,7 @@ const SourceTabs = React.createClass({
       label: "Show source",
       accesskey: "s",
       disabled: false,
-      click: () => showSource(tab)
+      click: () => showSource(tab, Math.random())
```

Comment:
Hi @jasonLaster 

I use this to fix "revealing the same source again to refocus it". The reason that it didn't refocus for revealing the same source was that `this.props.shownSource` did not change in `SourcesTree.js` when we revealed the same source. The `tab` that we passed into `showSource` in `SourceTabs.js` was the same. In order to change `this.props.shownSource`  in `SourcesTree.js` which would call `componentWillReceiveProps`, I used a random number so that every time a user clicks on reveal `this.props.shownSource` is updated. Based on [StackOverflow](http://stackoverflow.com/questions/28199100/probability-of-getting-the-same-value-using-math-random), the probability of having a collision is 60 million, so it should be safe to use. 

I was only able to come up with this solution, so I'm not sure if there is other way that we can do to call `componentWillReceiveProps` in `SourcesTree` when we click on reveal of the same source.